### ISSUE TITLE
client find () method problem

### ISSUE THREAD

**Issue by GlauberF** [feathersjs/feathers#711]
**client find () method problem**

I've been working all day to understand the problem and I do not know what it is.
using the find () method on the client, sometimes the data comes, sometimes the error comes.
And if I use the $ http, normal at all times comes the data.

see the image, use in the first call the $ http method of angular and then call find () method.

![captura de tela de 2017-10-27 09-36-18](https://user-images.githubusercontent.com/11314585/32102636-f8ba74c4-bafb-11e7-95cd-6cbf7ed6bb5d.png)

In $ http, returns the right data, but in the find of this error.

The funniest thing is that sometimes the find () method works and sometimes it does not.

**console debug when error**

![captura de tela de 2017-10-27 09-59-38](https://user-images.githubusercontent.com/11314585/32102954-39159386-bafd-11e7-889b-974d4dddbf15.png)

I noticed that the error usually happens, when F5 is given in the browser.

**Comment by daffl** [comment]
This might be an issue with https://github.com/beevelop/ng-feathers which you are using. Try using the Feathers client directly to verify first.

**Comment by jakobvogt** [comment]
**I've got the same problem using the official Feathers client** and Angular 4.

So far I've only built projects where I authenticate before interacting with the backend.
Today I started a project where I directly do a *watch().find()* as soon as the user visits the page,
but for some reason, I sometimes get an *Error: 'Timeout of 5000ms'* (the same error as GlauberF).
So my first thought was that the socket connection probably wasn't established at the time calling because when I did it with the *setTimeout(..., 1000)* function, it worked.

The strange thing is that **it doesn't happen all the time**. 
When I, for example, add an *"authenticate('jwt')"* as before hook, it doesn't throw a timeout error but the expected 'no auth token'-Error.

**So my question:** Is there some kind of callback or promise for when the connection is established that I need to use or is this a bug?

**Comment by jakobvogt** [comment]
After 5 days of trying to find the problem and nearly losing my mind, I finally found the solution.
It was a missing dependency... I don't know why there weren't any error messages beforehand.
The thing that still confuses me is that I've built several projects without having any issues regarding the missing dependency.
**Fix:**
Add these two lines to the dependencies in the package.json file
```
"socket.io-client": "^2.0.4",
"@types/socket.io-client": "^1.4.31"
```
